### PR TITLE
Feature/disable talk chat notifications

### PR DESCRIPTION
**Pull request by claucambra**
Screenshot 2023-02-13 at 14 50 10

### CODE REVIEW

File: src/libsync/configfile.h [hunk 1]
```diff
@@ -156,8 +156,11 @@ class OWNCLOUDSYNC_EXPORT ConfigFile
     [[nodiscard]] bool optionalServerNotifications() const;
     void setOptionalServerNotifications(bool show);
 
+    [[nodiscard]] bool showChatNotifications() const;
+    void setShowChatNotifications(bool show);
+
     [[nodiscard]] bool showCallNotifications() const;
-    void setShowCallNotifications(bool show);
+    void setShowCallNotifications(const bool show);
```
Comment (mgallien) [inline src/libsync/configfile.h:163]: why one has `const` for the single argument and not the other one ?
method declaration with plain old data argument should be without `const` and if deemed useful, implementation could add `const`
with POD data types, `const` is not part of the method signature so `const` if needed by the implementation should not appear in the method declaration

**Review by mgallien — CHANGES_REQUESTED** [review]
can you have a look at my comment, rebase and fix the conflict ?

File: src/libsync/configfile.h (hunk 1 above)
Comment (claucambra) [reply src/libsync/configfile.h]: Thanks for the explanation, TIL

fixed now

**Comment by claucambra** [comment]
Thanks for review @mgallien addressed comments

File: src/gui/tray/usermodel.cpp [hunk 2]
```diff
@@ -183,7 +183,12 @@ void User::showDesktopTalkNotification(const Activity &activity)
 {
     const auto notificationId = activity._id;
 
-    if (!canShowNotification(notificationId)) {
+    const ConfigFile cfg;
+    const auto userStatus = _account->account()->userStatusConnector()->userStatus().state();
+    if (!canShowNotification(notificationId) ||
+        userStatus == OCC::UserStatus::OnlineStatus::DoNotDisturb ||
+        !cfg.showChatNotifications()) {
+
```
Comment (mgallien) [inline src/gui/tray/usermodel.cpp:191]: the local check of DnD mode is clearly duplicating a check made server side
is there any reason to add it there ?
seems like redundant and adding one more check in the if for nothing

Comment (claucambra) [reply src/gui/tray/usermodel.cpp]: I was likely being cautious here

I removed the check

**Review by mgallien — APPROVED** [review]

**Pull request merged**